pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessor.java

```diff
@@ -104,8 +104,15 @@ public void process()
       // Update single-column indices, like inverted index, json index etc.
       IndexCreatorProvider indexCreatorProvider = IndexingOverrides.getIndexCreatorProvider();
       for (ColumnIndexType type : ColumnIndexType.values()) {
-        IndexHandler handler = IndexHandlerFactory.getIndexHandler(type, _segmentMetadata, _indexLoadingConfig);
+        IndexHandler handler =
+            IndexHandlerFactory.getIndexHandler(type, _segmentMetadata, _indexLoadingConfig, _schema);
         handler.updateIndices(segmentWriter, indexCreatorProvider);
+        if (type == ColumnIndexType.FORWARD_INDEX) {
```

Comment:
Rather than relying on ColumnIndexType enum entry order, I feel this code should explicitly ensure that FORWARD_INDEX handler is the first handler to be invoked and then everyone else in the loop

We have seen cases (in the recent past) where either people disregarded or just simply forgot to honor the javadoc guideline regarding enum entry order. So, code should be more resilient against such cases

Reply:
Ack. Can we take this up as a TODO? I have a few ideas and we can discuss this offline to refactor the code.

Reply:
Yes TODO is fine.